tests/scene/test_http_request.h

```diff
@@ -0,0 +1,168 @@
+/*                             GODOT ENGINE                               */
+/*                        https://godotengine.org                         */
+/**************************************************************************/
+/* Copyright (c) 2014-present Godot Engine contributors (see AUTHORS.md). */
+/* Copyright (c) 2007-2014 Juan Linietsky, Ariel Manzur.                  */
+/*                                                                        */
+/* Permission is hereby granted, free of charge, to any person obtaining  */
+/* a copy of this software and associated documentation files (the        */
+/* "Software"), to deal in the Software without restriction, including    */
+/* without limitation the rights to use, copy, modify, merge, publish,    */
+/* distribute, sublicense, and/or sell copies of the Software, and to     */
+/* permit persons to whom the Software is furnished to do so, subject to  */
+/* the following conditions:                                              */
+/*                                                                        */
+/* The above copyright notice and this permission notice shall be         */
+/* included in all copies or substantial portions of the Software.        */
+/*                                                                        */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#ifndef TEST_HTTP_REQUEST_H
+#define TEST_HTTP_REQUEST_H
+
+#include "scene/main/http_request.h"
+#include "tests/core/io/test_http_client_mock.h"
+#include "tests/test_macros.h"
+
+#include "thirdparty/cpp_mock/cpp_mock.h"
+
+namespace TestHTTPRequest {
+
+static inline Array build_array() {
+	return Array();
+}
+template <typename... Targs>
+static inline Array build_array(Variant item, Targs... Fargs) {
+	Array a = build_array(Fargs...);
+	a.push_front(item);
+	return a;
+}
+
+static inline PackedStringArray build_headers() {
+	return PackedStringArray();
+}
+template <typename... Targs>
+static inline PackedStringArray build_headers(Variant item, Targs... Fargs) {
+	PackedStringArray psa = build_headers(Fargs...);
+	psa.push_back(item);
+	return psa;
+}
+
+TEST_CASE("[Network][HTTPRequest] Download chunk size is set when HTTP client is disconnected") {
+	HTTPClientMock::make_current();
+	HTTPRequest *http_request = memnew(HTTPRequest);
+	auto http_client = HTTPClientMock::current_instance;
```

Comment:
`auto` is [not allowed](https://docs.godotengine.org/en/stable/contributing/development/cpp_usage_guidelines.html#auto-keyword)